include/zephyr/bluetooth/l2cap.h

```diff
@@ -427,6 +427,8 @@ struct bt_l2cap_chan_ops {
  */
 #define BT_L2CAP_SDU_CHAN_SEND_RESERVE (BT_L2CAP_SDU_BUF_SIZE(0))
 
+struct bt_l2cap_server;
```

Comment:
@donatieng is this really needed given that you use the struct inside the actual declaration? Can you try without?